src/controller/expenses.controller.js

```diff
@@ -0,0 +1,122 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = async (req, res) => {
+  const { userId, from, to, categories } = req.query;
+  const expenses = await expensesService.getAllExpenses();
+
+  if (!userId && !categories && (!from || !to)) {
+    return res.send(expenses);
+  }
+
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(
+      filteredExpenses,
+      userId,
+    );
+  }
+
+  if (categories) {
+    const normalizedCategories = expensesService.normalizeCategory(categories);
+
+    filteredExpenses = expensesService.filterExpensesByCategory(
+      filteredExpenses,
+      normalizedCategories,
+    );
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(
+      filteredExpenses,
+      fromDate,
+      toDate,
+    );
+  }
+
+  res.send(filteredExpenses);
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+
+  const expense = await expensesService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const create = async (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId || !spentAt || !title || !amount) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const user = await userService.getById(userId);
```

Comment:
This line queries the database for the user associated with the provided userId. However, if the user does not exist, the code will send a 400 status code. It would be more appropriate to send a 404 status code in this case, as the user resource was not found.